docs/wiki/research/scaling/eip-4844.md

```diff
@@ -0,0 +1,220 @@
+- execute transactions inside the block payload
+- update the state, which is the result of executing transactions
+- build the new block
+- return the new block to the CL
+
+With EIP-4844, the `process_execution_payload` adds the parameter `versioned_hashes` to be passed to the `verify_and_notify_new_payload` function.
+
+`versioned_hashes` is an array of [hashes](https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/beacon-chain.md#modified-process_execution_payload) for each blob of KZG Committment, which are a type of cryptographic commitment particularly useful for their efficiency in creating and verifying proofs of data availability and correctness.
+
+KZG commitments provide the ability to prove that specific pieces of data are included in the set without revealing the entire dataset. This is particularly useful for scalability solutions because it does not require for every node to store the whole blockchain to prove transactions correcteness.
+
+#### 3. New Block Header checks
+
+See go-ethereum file "consensus.go" for checks added to the block header:
+
+- [link](https://github.com/ethereum/go-ethereum/blob/0a2f33946b95989e8ce36e72a88138adceab6a23/consensus/beacon/consensus.go#L274)
+
+### Changes on the Execution Specs
+
+The upgrade which will introduce EIP-4844 into the Execution Layer has been labeled **Cancun**.
+
+#### 1. Add checks inside the State Transition Function
+
+The EL now must check that the blob specific fields are valid for each transaction that is going to be executed in the State Transition Function (STF).
+
+The checks include (For the specs code, see [here](https://eips.ethereum.org/EIPS/eip-4844#execution-layer-validation)):
+
+- check that the signer has enough balance to cover the cost of both transaction gas fee and blob gas fees
+
+```python
+# modify the check for sufficient balance
+max_total_fee = tx.gas * tx.max_fee_per_gas
+if get_tx_type(tx) == BLOB_TX_TYPE:
+    max_total_fee += get_total_blob_gas(tx) * tx.max_fee_per_blob_gas
+assert signer(tx).balance >= max_total_fee
+```
+
+- check that the blob transaction contains at least 1 valid `blob_versioned_hash` (see CL changes) and that they are formatted correctly
+
+```python
+assert len(tx.blob_versioned_hashes) > 0
+```
+
+- check that the user is willing to pay at least the current blob base fee
+
+```python
+assert tx.max_fee_per_blob_gas >= get_blob_base_fee(block.header)
+```
+
+Finally, the EL STF must keep track of the gas being gas used for blob transactions (same as it already happens for EIP1559 transactions).
+
+```python
+blob_gas_used += get_total_blob_gas(tx)
+```
+
+## A note about previous attempts/research to improve scalability for rollups
+
+To accomplish scalability, the idea was initially to decrease transaction `calldata` gas cost with [EIP-4488](https://eips.ethereum.org/EIPS/eip-4488). However, this EIP is now irrelevant because EIP-4844 has been implemented instead. Here is a brief introduction for the EIP-4488.
```

Comment:
there was also 4490 which is the same proposal with slightly different parameters, might be worth mentioning for extra insights

Reply:
I added EIP-4490 [here](https://github.com/eth-protocol-fellows/protocol-studies/pull/51/commits/ad9422b9a3aa6bb510f0fd76f2719bba0358bfdb)